content/filterx/function-reference.md

```diff
@@ -0,0 +1,406 @@
+---
+title: Filterx function reference
+linkTitle: Functions
+weight: 2500
+---
+
+<!-- This file is under the copyright of Axoflow, and licensed under Apache License 2.0, except for using the Axoflow and AxoSyslog trademarks. -->
+
+{{< include-headless "chunk/filterx-experimental-banner.md" >}}
+
+This page describes the functions you can use in [filterx blocks]({{< relref "/filterx/_index.md" >}}).
+
+## cache_json_file {#cache-json-file}
+
+Load the contents of an external JSON file in an efficient manner. You can use this to lookup contextual information. (Basically, this is a filterx-specific implementation of the [`add-contextual-data() functionality`]({{< relref "/chapter-enrich-data/data-enrichment-add-contextual-data/_index.md" >}}).)
+
+Usage: `cache_json_file("/path/to/file.json")`
+
+For example, if your `context-info-db.json` file contains the following:
+
+```shell
+{
+  "nginx": "web",
+  "httpd": "web",
+  "apache": "web",
+  "mysql": "db",
+  "postgresql": "db"
+}
+```
+
+Then the following filterx expression selects only "web" traffic:
+
+```shell
+filterx {
+  declare known_apps = cache_json_file("/context-info-db.json");
+  ${app} = known_apps[${PROGRAM}] ?? "unknown";
+  ${app} == "web";  # drop everything that's not a web server log
+}
+```
+
+{{% alert title="Note" color="info" %}}
+{{< product >}} reloads the contents of the JSON file only when the {{< product >}} configuration is reloaded.
+{{% /alert %}}
+
+## datetime
+
+Cast a value into a datetime variable.
+
+Usage: `datetime(<string or expression to cast as datetime>)`
+
+For example:
+
+```shell
+date = datetime("1701350398.123000+01:00")
+```
+
+Usually, you use the [strptime](#strptime) filterx function to create datetime values. Alternatively, you can cast an integer, double, string, or isodate variable into datetime with the `datetime()` filterx function. Note that:
+
+- When casting from an integer, the integer is the number of seconds elapsed since the UNIX epoch (January 1, 1970 12:00:00 AM).
+- When casting from a double, the double is the number of milliseconds elapsed since the UNIX epoch (January 1, 1970 12:00:00 AM).
```

Comment:
```suggestion
- When casting from an integer, the integer is the number of nanoseconds elapsed since the UNIX epoch (January 1, 1970 12:00:00 AM).
- When casting from a double, the double is the number of seconds elapsed since the UNIX epoch (January 1, 1970 12:00:00 AM).
```